rpc/src/v1/impls/eth_pubsub.rs

```diff
@@ -172,6 +178,21 @@ impl<C> ChainNotificationHandler<C> {
 		}
 	}
 
+	fn notify_return_data<F>(&self, enacted: &[H256], calculate_return_data: F) where
+		F: Fn(&[H256]) -> Vec<ReturnData>
+	{
+		let return_data = calculate_return_data(enacted);
+
+		for subscriber in self.return_data_subscribers.read().values() {
+			let remote = self.remote.clone();
+			let subscriber = subscriber.clone();
+
+			for return_datum in return_data.clone() {
```

Comment:
You're definitely correct that `remote` and `subscriber` didn't need to be cloned. I believe `return_data` does, however, because the same return data is sent to all subscribers.